ui/imports/shared/stores/RootStore.qml

```diff
@@ -36,7 +36,8 @@ QtObject {
 
     property var history: walletSectionTransactions
     property var historyTransactions: walletSectionTransactions.model
-    property bool isNonArchivalNode:  history.isNonArchivalNode
+    property bool isNonArchivalNode: typeof history !== "undefined" ? history.isNonArchivalNode
```

Comment:
It can still be undefined

Reply:
Yes, but `history.isNonArchivalNode` will never be evaluated. 

```qml
property var history
property bool isNonArchivalNode: history ? history.isNonArchivalNode : false
```

Just double-checked the snippet above. Even though `history` is undefined there is no warning.

```qml
// property var history
property bool isNonArchivalNode: history ? history.isNonArchivalNode : false
```

This would result in a warning indeed but not the case here.